src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -65,14 +65,17 @@ parseTypeSynonymDeclaration =
 
 parseValueDeclaration :: TokenParser Declaration
 parseValueDeclaration = do
-  name <- parseIdent
-  binders <- P.many parseBinderNoParens
-  value <- Left <$> (indented *>
-                       P.many1 ((,) <$> parseGuard
-                                    <*> (indented *> equals *> parseValueWithWhereClause)
-                               ))
-       <|> Right <$> (indented *> equals *> parseValueWithWhereClause)
-  return $ ValueDeclaration name Public binders value
+  binders <- (:) <$> parseBinderAtom <*> (P.many parseBinderNoParens)
+  case binders of
+    VarBinder name : binders' -> do
+      value <- Left <$> (indented *>
+                          P.many1 ((,) <$> parseGuard
+                                        <*> (indented *> equals *> parseValueWithWhereClause)
+                                  ))
+          <|> Right <$> (indented *> equals *> parseValueWithWhereClause)
+      return $ ValueDeclaration name Public binders' value
+    [binder] -> BoundValueDeclaration binder <$> (indented *> equals *> parseValueWithWhereClause)
```

Comment:
It silently parses without error. How about separate `parseValueDeclaration` and `parseLocalValueDeclaration`? For the top level, we can just parse ident at the beginning of the decl, and for the local one, we can parse `parseBinder` as you commented above. I will check if it's possible.